manifests/proxydhcp.pp

```diff
@@ -3,7 +3,7 @@
   # puppet fact names are converted from ethX.X and ethX:X to ethX_X
   # so for alias and vlan interfaces we have to modify the name accordingly
   $interface_fact_name = regsubst($foreman_proxy::dhcp_interface, '[.:]', '_')
-  $ip   = pick($::foreman_proxy::dhcp_pxeserver, inline_template("<%= scope.lookupvar('::ipaddress_${interface_fact_name}') %>"))
+  $ip = pick_default($::foreman_proxy::dhcp_pxeserver, fact("ipaddress_${interface_fact_name}"))
   if ! is_ip_address($ip) {
```

Comment:
While you're at it, mind replacing this with the non-deprecated version to remove a deprecation warning? I think ```unless $ip =~ Stdlib::Compat::Ipv4 {``` should work. Same for the other 3 occurrences.